lib/addressable/idna/pure.rb

```diff
@@ -66,7 +66,7 @@ module IDNA
     # domain name as described in RFC 3490.
     def self.to_ascii(input)
       input = input.to_s unless input.is_a?(String)
-      input = input.dup
+      input = input.dup.force_encoding(Encoding::UTF_8).unicode_normalize(:nfkc)
```

Comment:
Should this still be NFKC?

Reply:
Good question and tricky answer. As far as I understand in [IDNA2008+UTC#46](https://unicode.org/reports/tr46/) there is the purely unicode normalization normalization step which is NFC, but there is also the IDNA mapping table, which is based on NFKC Casefold transformation. So in the end it does look similar to NFKC with some variations.

In IDNA2003 the "normalization" step was called "STRINGPREP" and was equivalent to NFKC (https://www.rfc-editor.org/rfc/rfc3491#ref-STRINGPREP):
> This profile specifies using Unicode normalization form KC, as described in [[STRINGPREP](https://www.rfc-editor.org/rfc/rfc3491#ref-STRINGPREP)].

Which is why [the libidn library only provides NFKC normalization functions](https://www.gnu.org/software/libidn/manual/libidn.html#Unicode-Normalization).

So in the end as we don't have the official mapping for the moment, NFKC is closer to the expected result (and was previously used), so I thought it would be better to keep it like that for the moment. This will probably change in #491 once we have the mapping table and all the conformance tests.